File: PoketDex/app/src/main/java/pt/rfsfernandes/model/pokemon/types/PokemonType.java
package pt.rfsfernandes.model.pokemon.types;

import com.google.gson.annotations.SerializedName;

import pt.rfsfernandes.model.SimpleModelData;

/**
 * Class PokemonType created at 1/16/21 15:34 for the project PoketDex
 * By: rodrigofernandes
 */
public class PokemonType {
  @SerializedName("slot")
  private int slot;

  @SerializedName("type")
  private SimpleModelData type;

  public PokemonType(int slot, SimpleModelData type) {
    this.slot = slot;
    this.type = type;
  }

  public int getSlot() {
    return slot;
  }

  public void setSlot(int slot) {
    this.slot = slot;
  }

  public SimpleModelData getType() {
    return type;
  }

  public void setType(SimpleModelData type) {
    this.type = type;
  }

  @Override
  public String toString() {
    return "PokemonType{" +
        "slot=" + slot +
        ", type=" + type +
        '}';
  }
}

File: PoketDex/app/src/main/java/pt/rfsfernandes/model/pokemon/types/PokemonType.kt
package pt.rfsfernandes.model.pokemon.types

import pt.rfsfernandes.model.SimpleModelData
import com.google.gson.annotations.SerializedName
import androidx.room.TypeConverters
import pt.rfsfernandes.data.local.PokemonTypeConverters

/**
 * Class PokemonType created at 1/16/21 15:34 for the project PoketDex
 * By: rodrigofernandes
 */
class PokemonType(
        @SerializedName("slot")
        var slot: Int,
        @TypeConverters(PokemonTypeConverters::class)
        @SerializedName("type")
        var type: SimpleModelData
        ) {

    override fun toString(): String {
        return "PokemonType{" +
                "slot=" + slot +
                ", type=" + type +
                '}'
    }
}